File: problem-16/prb16.c
#include <stdio.h>

int main()
{
    int d,n;
    scanf("%d",&d);
    if(d<0)
    {
        printf("Please enter positive number");
    }
    else if(d==0)
    {
        printf("Square root of %d is 0",d);
    }
    else if(d==1)
    {
        printf("Square root of %d is 1",d);
    }
    else
    {
        scanf("%d",&n);
        if(n==0)
        {
            printf("Invalid input");
        }
        else
        {
            if((d/n==n)&&(d%n==0))
            {
                printf("Square root of %d is %d",d,n);
            }
            else
            {
                printf("%d is not a square root of %d",n,d);
            }
        }
    }
    return 0;
}

File: problem-16/prb16.py
n1 = int(input())
if(n1 < 0):
    print("Please enter a positive number")
elif(n1 == 0):
    print("Square root of 0 is 0")
elif(n1 == 1):
    print("Square root of 1 is 1")
else:
    n2 = int(input())
    if(n2 == 0):
        print("Invalid Input")
    else:
        if((n1 / n2 == n2) and (n1 % n2 == 0)):
            print("Square root of " + str(n1) + " is " + str(n2))
        else:
            print(str(n2) + " is not a square root of " + str(n1))
